virttest/qemu_storage.py

```diff
@@ -986,9 +986,13 @@ def convert(self, params, root_dir, cache_mode=None,
         convert_cmd = self.image_cmd + " " + \
             self._cmd_formatter.format(self.convert_cmd, **cmd_dict)
 
-        LOG.info("Convert image %s from %s to %s", self.image_filename,
-                 self.image_format, convert_image.image_format)
-        process.run(convert_cmd)
+        timeout = convert_params.get_numeric("image_conversion_timeout", -1)
+        timeout = None if timeout == -1 else timeout
```

Comment:
Then maybe it seems to be the time to introduce a dedicated method to `Param` for getting timeouts, how does that sound? And later we may consider to add more features to support:

* setting timeouts in different units of time (e.g. `5m`, `1h`)
* basic calculations (with constants, e.g. `TEST_TIMEOUT * 0.8`)

Reply:
Hmm, I am not sure how `Params` object enhancements relates to the current pull request which aims specifically at `qemu-img` but bringing myself to talk about this new topic I would still say I am not sure whether introducing such methods in `Params` is not too data specific. The current type-handling methods are general enough and a small easy to maintain selection. Returning -1 is a very easy way to handle not just timeouts but various numeric parameters that might end up including infinity and the above is a simple solution that only uses -1 default value, something I hardly see as needing an entire new method with lots of added assumptions.

Reply:
> Returning -1 is a very easy way to handle not just timeouts but various numeric parameters that might end up including infinity and the above is a simple solution that only uses -1 default value, something I hardly see as needing an entire new method with lots of added assumptions.

Then, almost all the cases of handling timeouts may just have like this two lines approach (let's recall the situation of handling numerics before having `get_numeric`). Anyway, I understand that it's not a must to this PR, so I'd leave the proposal for future discussion, thanks.